mypy/semanal.py

```diff
@@ -719,7 +721,8 @@ def unbind_class_type_vars(self) -> None:
     def analyze_class_decorator(self, defn: ClassDef, decorator: Expression) -> None:
         decorator.accept(self)
 
-    def calculate_abstract_status(self, typ: TypeInfo) -> None:
+    @staticmethod
+    def calculate_abstract_status(typ: TypeInfo) -> None:
```

Comment:
You don't need to sell that to me, I completely agree :) I have used staticmethod trying to reduce the footprint of these changes, but this is pretty hopeless (and maybe not desirable, given this _is_ a refactoring PR).